build_config/test/build_config_test.dart

```diff
@@ -223,170 +223,106 @@ void expectPostProcessBuilderDefinitions(
-      equals(_expected.buildExtensions).matches(item.buildExtensions, _) &&
-      equals(_expected.requiredInputs).matches(item.requiredInputs, _) &&
-      equals(_expected.runsBefore).matches(item.runsBefore, _) &&
-      equals(_expected.appliesBuilders).matches(item.appliesBuilders, _) &&
-      new _TargetBuilderConfigDefaultsMatcher(_expected.defaults)
-          .matches(item.defaults, _) &&
-      item.autoApply == _expected.autoApply &&
-      item.isOptional == _expected.isOptional &&
-      item.buildTo == _expected.buildTo &&
-      item.import == _expected.import &&
-      item.key == _expected.key &&
-      item.package == _expected.package;
-
-  @override
-  Description describe(Description description) =>
-      description.addDescriptionOf(_expected);
-}
-
-class _PostProcessBuilderDefinitionMatcher extends Matcher {
-  final PostProcessBuilderDefinition _expected;
-  _PostProcessBuilderDefinitionMatcher(this._expected);
-
-  @override
-  bool matches(item, _) =>
-      item is PostProcessBuilderDefinition &&
-      equals(_expected.builderFactory).matches(item.builderFactory, _) &&
-      new _TargetBuilderConfigDefaultsMatcher(_expected.defaults)
-          .matches(item.defaults, _) &&
-      item.import == _expected.import &&
-      item.key == _expected.key &&
-      item.package == _expected.package;
-
-  @override
-  Description describe(Description description) =>
-      description.addDescriptionOf(_expected);
-}
-
-class _GlobalBuilderConfigMatcher extends Matcher {
-  final GlobalBuilderConfig _expected;
-  _GlobalBuilderConfigMatcher(this._expected);
-
-  @override
-  bool matches(item, _) =>
-      item is GlobalBuilderConfig &&
-      equals(_expected.options.config).matches(item.options.config, _) &&
-      equals(_expected.devOptions.config).matches(item.devOptions.config, _) &&
-      equals(_expected.releaseOptions.config)
-          .matches(item.releaseOptions.config, _);
-
-  @override
-  Description describe(Description description) =>
-      description.addDescriptionOf(_expected);
-}
-
-class _TargetBuilderConfigDefaultsMatcher extends Matcher {
-  final TargetBuilderConfigDefaults _expected;
-  _TargetBuilderConfigDefaultsMatcher(this._expected);
-
-  @override
-  bool matches(item, _) =>
-      item is TargetBuilderConfigDefaults &&
-      equals(_expected.generateFor.include)
-          .matches(item.generateFor.include, _) &&
-      equals(_expected.generateFor.exclude)
-          .matches(item.generateFor.exclude, _) &&
-      equals(_expected.options.config).matches(item.options.config, _) &&
-      equals(_expected.devOptions.config).matches(item.devOptions.config, _) &&
-      equals(_expected.releaseOptions.config)
-          .matches(item.releaseOptions.config, _);
-
-  @override
-  Description describe(Description description) =>
-      description.addDescriptionOf(_expected);
-}
+Matcher _matchesBuilderDefinition(BuilderDefinition definition) =>
```

Comment:
At what point do you just implement `==` on BuilderDefinition?

Reply:
The failure message isn't as nice.

I could be convinced that since these are data types they should implement equals anyway... but I'm not sure.

Reply:
It would also have to get into using DeepEquality and stuff